packages/core/src/entity/ArrayCollection.ts

```diff
@@ -337,6 +337,13 @@ export class ArrayCollection<T extends object, O extends object> {
     return true;
   }
 
+  first(): T | undefined {
+    if (this.isInitialized()) {
+      return this[0];
+    }
+    return undefined;
```

Comment:
the method should throw if the collection is not initialized, same for the `last` one. i would move the `checkInitialized` to ArrayCollection (and mark it as protected) and do this instead
```suggestion
    this.checkInitialized();
    return this[0];
```

then the `Collection` implementations can be removed completely

Reply:
`checkInitialized` doesnt exist on ArrayCollection class.  I have the checkInitialized on the collection one though
```ts
  override first(): T | undefined {
    this.checkInitialized();
    return super.first();
  }

```